src/main/java/com/dataloom/neuron/signals/Signal.java

```diff
@@ -17,31 +17,63 @@
  * You can contact the owner of the copyright at support@thedataloom.com
  */
 
-package com.dataloom.neuron;
+package com.dataloom.neuron.signals;
+
+import java.util.List;
+import java.util.UUID;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.dataloom.authorization.Principal;
 import com.dataloom.client.serialization.SerializationConstants;
+import com.dataloom.neuron.SignalType;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Optional;
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
 public class Signal {
 
     private static final Logger logger = LoggerFactory.getLogger( Signal.class );
 
-    private SignalType type;
+    private SignalType       type;
+    private List<UUID>       aclKey;
+    private Principal        principal;
+    private Optional<String> details;
 
     @JsonCreator
-    public Signal( @JsonProperty( SerializationConstants.TYPE_FIELD ) SignalType type ) {
+    public Signal(
+            @JsonProperty( SerializationConstants.TYPE_FIELD ) SignalType type,
+            @JsonProperty( SerializationConstants.ACL_KEY ) List<UUID> aclKey,
+            @JsonProperty( SerializationConstants.PRINCIPAL ) Principal principal,
+            @JsonProperty( SerializationConstants.DETAILS_FIELD ) Optional<String> details ) {
```

Comment:
Are you referring to `Optional<String> details`?

Actually, now that you mention arbitrary objects, I wanted to talk to you about how that would work. I wasn't sure how to support storing `Signal`s with arbitrary properties, so I decided to start out with just a simple optional String.